#!/usr/bin/env python
PACKAGE = "fetch_open_auto_dock"

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, int_t, double_t, bool_t

gen = ParameterGenerator()

gen.add("k1", double_t, 0, "ratio in change of theta to rate of change in r", 0.0, 1.0, 30.0)
gen.add("k2", double_t, 0, "speed at which we converge to slow system", 0.0, 3.0, 30.0)
gen.add("min_velocity", double_t, 0, "min velocity", 0.0, 0.10, 1.0)
gen.add("max_velocity", double_t, 0, "max velocity", 0.0, 0.15, 2.0)
gen.add("max_angular_velocity", double_t, 0, "max angular velocity", 0.0, 0.5, 3.0)
gen.add("beta", double_t, 0, "how fast velocity drops as k increases", 0.0, 0.2, 4)
gen.add("lambda", double_t, 0, "lambda", 0.0, 2.0, 10.0)
gen.add("dist", double_t, 0, "used to create the tracking line", 0.0, 0.4, 4)

exit(gen.generate(PACKAGE, "fetch_open_auto_dock", "FetchControllerGain"))
